### DISCUSSION TITLE
relationships resolution

### DISCUSSION THREAD

**Discussion by kivS** [pocketbase/pocketbase#155] [Q&A] [answered]
Hi there, I'm trying to understand a bit about how the relationships are implemented in the project.
From what I can see the generated tables are using array fields on the records and not using foreign keys to maintain relationships.

I'm curious to hear what was the reasoning behind it and how are the relationships being resolved, is it done with sqlite json functions and operators or is it from the Go side of things?
say, I want to use raw sqls outside of PocketBase, any considerations to have in mind?

Cheers

**Comment by ganigeorgiev** [comment, marked as answer]
There are no sqlite maintained foreign keys, because the user is allowed to change dynamically the related collection behavior (or entirely to change the collection) through the field options.

The relationships in PocketBase are resolved on app level as regular joins as in any other application and depending on the relation field option `MaxSelect` (whether is single or multiple) we are using string and json array to store the id(s) of the related records (see [record_field_resolver](https://github.com/pocketbase/pocketbase/blob/a16b0c900420cd70b0c79d4ac824a61a1e02585d/resolvers/record_field_resolver.go#L158-L183)).

So in other words, the only major difference in handling relationships from other applications compared to PocketBase is that the foreign key constraints (eg. cascade) are implemented on app level.

_One other minor difference is that, when you create a standard foreign key in a table, usually it also creates an index for that column. Currently in PocketBase you cannot define indexes through the Admin UI for individual fields (aka. columns), but that is in the [roadmap](https://github.com/orgs/pocketbase/projects/2)._

**Reply by kivS** [reply]
Thanks for the answer
